### ISSUE TITLE
Is anyone get a files.json dowload link from {{url}}?

### ISSUE THREAD

**Issue by FridaG** [tomitrescak/meteor-uploads#45]
If I upload an image or an pdf file, everything works fine.  But if I upload a different kind of file, like a .docx, and use {{file.url}} to link to the file, even though it says the correct url, when I click to download, it downloads a "files.json" file. If I use {{file.path}} to link to the file, I get an "unexpected error" message at the link.  Has this happened to anyone else?  I can see in my database that both the absolute url and the path are correct, so I'm wondering if this is an issue related to file type.  

update: I changed the extension from json back to docx, and low and behold, it was my file.  So now I am totally miffed, I checked the response headers and they looked correct...really don't understand why it changed the name and extension of my file

**Comment by tomitrescak** [comment]
Friday will check this asap.

**Comment by tomitrescak** [comment]
FridaG, I checked it today and I got the standard behaviour and can download anything I can. I have created a small test project which you can find here [https://dl.dropboxusercontent.com/u/3418607/UploadTest.zip](https://dl.dropboxusercontent.com/u/3418607/UploadTest.zip). Try if it is doing the same.

**Comment by FridaG** [comment]
Tomitrescak, I hope you know how much I appreciate the work and help you've put into this project, I will definitely be passing the help on as much as I can.  What browser are you using?  I tried your test project, but it gave me the same behavior I was receiving: a .doc gave me a files.json in both mozilla firefox and chrome, but I can change the extension of the file after download and it is correct.  

Do you think sending an application/octet-stream header would sort it out?

**Comment by FridaG** [comment]
![screen shot 2015-02-20 at 11 15 43 pm](https://cloud.githubusercontent.com/assets/5955930/6286741/7d0a31ba-b956-11e4-90dc-b5b781927cfc.jpg)
Here are my headers.  of note is that when I have this issue with the files.json, I get three GET requests, wheras I only have one for files like pdfs where it works perfectly:

![screen shot 2015-02-20 at 11 15 26 pm](https://cloud.githubusercontent.com/assets/5955930/6286763/ab88a274-b956-11e4-983b-fb2c71f3bc9f.jpg)

**Comment by FridaG** [comment]
and here with a .docx (please don't waste your time if you think this is just an issue with my personal hardware!)  
![screen shot 2015-02-20 at 11 19 39 pm](https://cloud.githubusercontent.com/assets/5955930/6286789/fe2d7086-b956-11e4-8f27-b34e867e6163.jpg)

info callback: 
![screen shot 2015-02-20 at 11 20 39 pm](https://cloud.githubusercontent.com/assets/5955930/6286799/22f92f18-b957-11e4-9789-d99bbc3567dc.jpg)

**Comment by FridaG** [comment]
last one: here's a working pdf example: 
![screen shot 2015-02-20 at 11 24 05 pm](https://cloud.githubusercontent.com/assets/5955930/6286842/9ed83908-b957-11e4-9fc8-6f56c895d476.jpg)

and the single GET request:
![screen shot 2015-02-20 at 11 23 56 pm](https://cloud.githubusercontent.com/assets/5955930/6286848/a7acb96e-b957-11e4-8912-8bdac375346c.jpg)

**Comment by tomitrescak** [comment]
Frida, version 1.2.0 solves these problems. I had to replace the whole file serving middleware since `connect` was acting up on me. Now I'm checking also on file existence and returning 404 when file does not exist or 403 when trying to get the directory or 500 when trying to go for symlinks. Also please note that server now supports these mime types:

```javascript
mimeTypes: {
    "html": "text/html",
    "jpeg": "image/jpeg",
    "jpg": "image/jpeg",
    "png": "image/png",
    "gif": "image/gif",
    "js": "text/javascript",
    "css": "text/css",
    "pdf": "application/pdf",
    "doc": "application/msword",
    "docx": "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
    "zip": "application/zip, application/x-compressed-zip",
    "txt": "text/plain"
  }
```
everything else is served as octet stream. I hope this will work for you, please close this issue if it will.

**Comment by FridaG** [comment]
<3!! you are a miracle worker, it is all sorted now.  yes, thank you very much.

**Comment by FridaG** [comment]
closed #45

**Comment by tomitrescak** [comment]
Glad I could help, I'll close this issue if you don't mind. Please reopen if necessary.
